### PR TITLE
Baseline analysis

### PR DESCRIPTION
**Pull request by tronxd**
Adds the `--baseline` option to use baseline file analysis 
Only new failed/pass checks with respect to the baseline are reported
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: checkov/common/output/baseline.py [hunk 1]
```diff
@@ -50,9 +53,27 @@ def to_dict(self) -> dict:
...
-    # @staticmethod
-    # def from_json(o: dict) -> "Baseline":
-    #     baseline = Baseline()
-    #     baseline.failed_checks = ...
-    #     return baseline
+    def compare_and_reduce_reports(self, scan_reports: List[Report]) -> None:
+        for scan_report in scan_reports:
+            scan_report.passed_checks = [check for check in scan_report.passed_checks if
+                                         self._is_check_in_baseline(check)]
+            scan_report.skipped_checks = []
```
Comment (nimrodkor) [inline checkov/common/output/baseline.py:60]: @tronxd This is actually very cool, what you did in passed checks. Why not do it for skipped checks as well?

File: checkov/common/output/baseline.py [hunk 2]
```diff
@@ -50,9 +53,27 @@ def to_dict(self) -> dict:
...
+        failed_check_resource = check.resource
+        for baseline_failed_check in self.failed_checks:
+            for finding in baseline_failed_check["findings"]:
+                if finding["resource"] == failed_check_resource and failed_check_id in finding["check_ids"]:
+                    return True
+
+    @staticmethod
+    def from_json(file_path: str) -> "Baseline":
+        baseline = Baseline()
+        baseline.path = file_path
```
Comment (nimrodkor) [inline checkov/common/output/baseline.py:76]: @tronxd Nice!

File: checkov/main.py [hunk 3]
```diff
@@ -129,27 +129,34 @@ def run(banner=checkov_banner, argv=sys.argv[1:]):
...
         return
 
+    baseline = None
+    if config.baseline:
+        baseline = Baseline.from_json(config.baseline)
+
     external_checks_dir = get_external_checks_dir(config)
     url = None
-
+    created_baseline_path = None
```
Comment (nimrodkor) [inline checkov/main.py:138]: @tronxd Why is it needed here?

**Review by nimrodkor — APPROVED** [review]
Very nice, minor comments only

File: checkov/common/output/baseline.py (hunk 1 above)
Comment (tronxd) [reply checkov/common/output/baseline.py]: Done

File: checkov/main.py (hunk 3 above)
Comment (tronxd) [reply checkov/main.py:138]: To pass it to runner/registry, if it's None nothing happens

**Pull request merged**